cmd/nerdctl/image/image_pull_linux_test.go

```diff
@@ -236,3 +236,44 @@ func TestImagePullSoci(t *testing.T) {
 
 	testCase.Run(t)
 }
+
+func TestImagePullProcessOutput(t *testing.T) {
+	nerdtest.Setup()
+
+	testCase := &test.Case{
+		Require: test.Require(
+			test.Linux,
+		),
+		SubTests: []*test.Case{
+			{
+				Description: "Pull Image - output should be in stdout",
+				Setup: func(data test.Data, helpers test.Helpers) {
+					helpers.Anyhow("rmi", "-f", testutil.BusyboxImage)
+				},
+				Command: func(data test.Data, helpers test.Helpers) test.TestableCommand {
+					return helpers.Command("pull", testutil.BusyboxImage)
+				},
+				Expected: func(data test.Data, helpers test.Helpers) *test.Expected {
+					return &test.Expected{
+						Output: test.Contains(testutil.BusyboxImage)}
+				},
+			},
+			{
+				Description: "Run Container with image pull - output should be in stderr",
+				Setup: func(data test.Data, helpers test.Helpers) {
+					helpers.Anyhow("rmi", "-f", testutil.BusyboxImage)
+				},
+				Command: func(data test.Data, helpers test.Helpers) test.TestableCommand {
+					return helpers.Command("run", "--rm", testutil.BusyboxImage)
+				},
+				Expected: func(data test.Data, helpers test.Helpers) *test.Expected {
```

Comment:
These are absolutely fine of course ^.

I thought I would mention though that there is also more compact notation when the test commands or conditions are really simple and you do not need access to data or helpers:
```go

Command: test.Command("run", "--rm", testutil.BusyboxImage),
Expected: test.Expects(0, nil, test.DoesNotContains(testutil.BusuboxImage)),
```